authentication/views.py

```diff
@@ -1,13 +1,57 @@
 from django.shortcuts import render
 from django.contrib.auth import login,logout,authenticate
-# Create your views here.
+from django.shortcuts import redirect
+from django.contrib.auth.models import User
 
 
 def loginView(request):
-    pass
+    if request.user.is_authenticated :
+        return redirect('index')
+    if request.method == "POST":
+        username = request.POST['username']
+        password = request.POST['password']
+        user = authenticate(request, username=username, password=password)
+        if user is not None:
+            login(request, user)
+            return redirect('index')
+        else:
+            return render(request, 'login.html', { 'msg': "Invalid Credentials" })
+    else:
+        return render(request, 'login.html')
 
 def logoutView(request):
-    pass
+    logout(request)
+    return redirect('index')
 
 def registerView(request):
-    pass
\ No newline at end of file
+    if request.method == "POST":
+        try:
+            username = request.POST['username']
+            firstname = request.POST['firstname']
+            lastname = request.POST['lastname']
+            password1 = request.POST['password1']
+            password2 = request.POST['password2']
+            email = request.POST['email']
+        except:
```

Comment:
As a good coding practice, whenever you use try-except block, capture only the exceptions which you want to catch (IndexError, IntegrityError, etc.)